packages/lib-core/src/runtime/coderefs.ts

```diff
@@ -126,18 +116,22 @@ export const decodeCodeRefs = (extension: LoadedExtension, entryModule: PluginEn
 };
 
 /**
- * In the extension's `properties` object, replace all {@link CodeRef} functions
- * with the corresponding values by resolving the associated Promises.
+ * In the extension's `properties` object, replace all {@link CodeRef} functions with
+ * the corresponding values by resolving the associated Promises.
+ *
+ * This is an asynchronous operation that completes when all of the associated Promises
+ * are either resolved or rejected. Each code reference resolution error will cause the
+ * associated value to be set to `undefined`.
  *
- * This is an asynchronous operation that completes when all of the associated
- * Promises are either resolved or rejected. If there were no resolution errors,
- * the resulting Promise resolves with the updated extension.
+ * The resulting Promise resolves with the updated extension; its `properties` object
+ * is cloned in order to preserve the original structure of decoded extensions.
  *
- * The updated extension is a new object; its `properties` are cloned in order
- * to preserve the original structure of decoded extensions.
+ * The resulting Promise never rejects. Use the `onResolutionErrors` callback to handle
```

Comment:
@vojtechszocs assuming this will need a major version bump as it is a breaking change?